### PR TITLE
Fix crash when creating partitioned table in utility mode

### PR DESCRIPTION
**Pull request by guofengrichard**
When transforming DISTRIBUTED BY clause in utility mode, NULL would be returned
as only QD can have policies. As a result, the child table in a partitioned
table would have NULL distributedBy, which would trigger assertion failure or
cause crash with current codes.

This patch fixes that by emitting an error message in that case to avoid the
crash.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/parser/parse_utilcmd.c [hunk 1]
```diff
@@ -416,6 +416,14 @@ transformCreateStmt(CreateStmt *stmt, const char *queryString, bool createPartit
 	{
 		int			numsegments = -1;
 
+		if (Gp_role != GP_ROLE_DISPATCH &&
+			!IsBinaryUpgrade &&
+			stmt->is_part_child)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_TABLE_DEFINITION),
+					 errmsg("child table in a partition created in utility "
```
Comment (zhangh43) [inline src/backend/parser/parse_utilcmd.c:424]: better to note the greenplum doesn't support create partition table in utility mode directly in error message.

Comment (zhangh43) [reply src/backend/parser/parse_utilcmd.c:424]: you can leave the reason 'doesn't have a policy' in the comment.

**Comment by ashwinstar** [comment]
:+1: Just note though that this might be temporary for master, as with PostgreSQL 12 merge we would be able to create partition table in utility mode as well. Not something recommended at all but will not crash or throw error like this PR is doing.

**Comment by guofengrichard** [comment]
> 👍 Just note though that this might be temporary for master, as with PostgreSQL 12 merge we would be able to create partition table in utility mode as well. Not something recommended at all but will not crash or throw error like this PR is doing.

That would be cool for master. Maybe we should target this PR for 6X (and 5X).

File: src/backend/parser/parse_utilcmd.c (hunk 1 above)
Comment (guofengrichard) [reply src/backend/parser/parse_utilcmd.c]: Thanks. Will do that.

**Pull request closed without merging**